pkg/provider/kubernetes/auth.go

```diff
@@ -66,8 +66,6 @@ func (c *Client) getAuth(ctx context.Context) (*rest.Config, error) {
 		if err != nil {
 			return nil, fmt.Errorf("could not fetch Auth.ServiceAccount: %w", err)
 		}
-	} else {
```

Comment:
This is not the correct fix. You need to put the below if statement into this.

Or rather, rewrite this whole else if section into a `switch` statement on Auth.Token, Auth.ServiceAccount and Auth.Cert. Then the default should be no auth provider given.

Reply:
ok had this in mind too, will refactor this